packages/venia-ui/lib/components/OrderHistoryPage/collapsedImageGallery.js

```diff
@@ -0,0 +1,49 @@
+import React, { useMemo } from 'react';
+import { useCollapsedImageGallery } from '@magento/peregrine/lib/talons/OrderHistoryPage/useCollapsedImageGallery';
+
+import { mergeClasses } from '../../classify';
+import defaultClasses from './collapsedImageGallery.css';
+import imageGalleryOperations from './collapsedImageGallery.gql';
+import Image from '../Image';
+
+const CollapsedImageGallery = props => {
+    const { items } = props;
+
+    const talonProps = useCollapsedImageGallery({
+        items,
+        ...imageGalleryOperations
+    });
+    const { imageData } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const remainderCount = items.length - 4;
+
+    const imageElements = useMemo(() => {
```

Comment:
What do you think about breaking this `useMemo` into 2 parts, 1 for items and other for the count if `remainderCount > 0`?

Not mandatory, you can omit this comment if you wish to.

Reply:
This shouldn't be needed, we are not over fetching.